File: app/src/main/java/gms/angus/angussoli/view/HighScoreFragment.java
package gms.angus.angussoli.view;

import android.app.Fragment;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
import com.google.android.gms.games.Games;

import java.util.List;

import butterknife.ButterKnife;
import butterknife.OnClick;
import gms.angus.angussoli.R;
import gms.angus.angussoli.model.RankedPlayer;

/**
 * Created by Harry Cliff on 11/28/17.
 */

public class HighScoreFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_high_score, container, false);
        ButterKnife.bind(this, view);

        // Inflate the layout for this fragment
        return view;
    }

    @OnClick(R.id.high_score)
    void onHighScoreClick(){
        Games.getLeaderboardsClient(getActivity(), getUserAccount()).getLeaderboardIntent(
                getString(R.string.highScore_board_id)).addOnSuccessListener(
                intent -> {
                    startActivityForResult(intent, 0 );
                });
    }

    @OnClick(R.id.total_score)
    void onTotalScoreClick(){
        Games.getLeaderboardsClient(getActivity(), getUserAccount()).getLeaderboardIntent(
                getString(R.string.totalScore_board_id)).addOnSuccessListener(
                intent -> {
                    startActivityForResult(intent, 0 );
                });
    }

    @OnClick(R.id.number_of_games)
    void onNumGamesClick(){
        Games.getLeaderboardsClient(getActivity(), getUserAccount()).getLeaderboardIntent(
                getString(R.string.numOfGame_board_id)).addOnSuccessListener(
                intent -> {
                    startActivityForResult(intent, 0 );
                });
    }

    @OnClick(R.id.average_game_pro_rank)
    void onAvgScoreClick(){
        getActivity().getFragmentManager().beginTransaction()
                            .replace(R.id.container, new ProScoreFragment(),
                                 ProScoreFragment.class.getSimpleName())
                            .addToBackStack(ToPlayFragment.class.getSimpleName()).commit();
    }

    @OnClick(R.id.achievements)
    void onAchievementsClick(){
        Games.getAchievementsClient(getActivity(), getUserAccount())
             .getAchievementsIntent()
             .addOnSuccessListener(
                     intent -> startActivityForResult(intent, 0));
    }

    private GoogleSignInAccount getUserAccount() {
        return ((GameActivity)getActivity()).userAccount;
    }

    private List<RankedPlayer> getRankedList(){
        return ((GameActivity)getActivity()).rankingLeaderboardScoreList;
    }
}

File: app/src/main/java/gms/angus/angussoli/view/HighScoreFragment.kt
package gms.angus.angussoli.view

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.google.android.gms.auth.api.signin.GoogleSignIn
import com.google.android.gms.auth.api.signin.GoogleSignInAccount
import com.google.android.gms.games.Games
import gms.angus.angussoli.R
import gms.angus.angussoli.databinding.FragmentHighScoreBinding

/**
 * Created by Harry Cliff on 11/28/17.
 */
class HighScoreFragment : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {


        // Inflate the layout for this fragment
        return FragmentHighScoreBinding.inflate(inflater, container, false).run {
            highScore.setOnClickListener{
                onHighScoreClick()
            }
            totalScore.setOnClickListener{
                onHighScoreClick()
            }
            numberOfGames.setOnClickListener{
                onHighScoreClick()
            }
            averageGameProRank.setOnClickListener{
                onAvgScoreClick()
            }
            achievements.setOnClickListener{
                onAchievementsClick()
            }
            root
        }
    }

    fun onHighScoreClick() {
        Games.getLeaderboardsClient(activity, GoogleSignIn.getLastSignedInAccount(activity)!!).getLeaderboardIntent(
            getString(R.string.leaderboard_highest_score)
        ).addOnSuccessListener { intent: Intent? -> startActivityForResult(intent, 0) }
    }

    fun onTotalScoreClick() {
        Games.getLeaderboardsClient(activity, GoogleSignIn.getLastSignedInAccount(activity)!!).getLeaderboardIntent(
            getString(R.string.leaderboard_total_score)
        ).addOnSuccessListener { intent: Intent? -> startActivityForResult(intent, 0) }
    }

    fun onNumGamesClick() {
        Games.getLeaderboardsClient(activity, GoogleSignIn.getLastSignedInAccount(activity)!!).getLeaderboardIntent(
            getString(R.string.leaderboard_number_of_games)
        ).addOnSuccessListener { intent: Intent? -> startActivityForResult(intent, 0) }
    }

    fun onAvgScoreClick() {
//        activity.fragmentManager.beginTransaction()
//            .replace(
//                R.id.container, ProScoreFragment(),
//                ProScoreFragment::class.java.simpleName
//            )
//            .addToBackStack(ToPlayFragment::class.java.simpleName).commit()
        Toast.makeText(activity, "Coming Soon", Toast.LENGTH_SHORT)
    }

    fun onAchievementsClick() {
        Games.getAchievementsClient(activity, GoogleSignIn.getLastSignedInAccount(activity)!!)
            .achievementsIntent
            .addOnSuccessListener { intent: Intent? -> startActivityForResult(intent, 0) }
    }
}